transport/transport.go

```diff
@@ -299,20 +299,18 @@ func (s *Stream) Read(p []byte) (n int, err error) {
 	return
 }
 
-type key int
-
 // The key to save transport.Stream in the context.
-const streamKey = key(0)
+type streamKey struct{}
```

Comment:
No, it has to be a distinct type. See `peer/peer.go` and `metadata/metadata.go` for other examples of doing this in the grpc code base.